src/core/meshcop/dataset_manager_ftd.cpp

```diff
@@ -224,35 +224,17 @@ otError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const Ip6:
     VerifyOrExit(!isUpdateFromCommissioner || type == Tlv::kPendingTimestamp || !doesAffectConnectivity,
                  state = StateTlv::kReject);
 
-    // update dataset
-    // Thread specification allows partial dataset changes for MGMT_ACTIVE_SET.req/MGMT_PENDING_SET.req
-    // from Commissioner.
-    // Updates based on existing active/pending dataset if it is from Commissioner.
     if (isUpdateFromCommissioner)
     {
-        // take active dataset as the update base for MGMT_PENDING_SET.req if no existing pending dataset.
-        if (type == Tlv::kPendingTimestamp && dataset.GetSize() == 0)
-        {
-            netif.GetActiveDataset().Get(dataset);
-        }
+        // Thread specification allows partial dataset changes for MGMT_ACTIVE_SET.req/MGMT_PENDING_SET.req
+        // from Commissioner based on existing active dataset.
+        netif.GetActiveDataset().Get(dataset);
```

Comment:
On L114 above, there is a call `mLocal.Get(dataset);`. Can we remove that line?

Reply:
Yes. Updated accordingly. Thanks.